k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -17,7 +17,7 @@
 import com.fsck.k9.mail.internet.SizeAware;
 import org.apache.commons.io.output.DeferredFileOutputStream;
 
-public class DecryptedTempFileBody extends BinaryAttachmentBody implements SizeAware {
+public class ProvidedTempFileBody extends BinaryAttachmentBody implements SizeAware {
```

Comment:
the 'temp' is incorrect, the body doesn't care about the nature of the file.maybe `FileProviderBackedBody`?